filesystems/s3/src/main/java/org/lerch/s3fs/S3FileSystemProvider.java

```diff
@@ -426,19 +422,141 @@ public void copy(Path source, Path target, CopyOption... options) throws IOExcep
             throw new FileAlreadyExistsException(format("target already exists: %s", target));
         }
 
-        String sourceBucketName = s3Source.getFileStore().name();
-        String keySource = s3Source.getKey();
-        String bucketNameTarget = s3Target.getFileStore().name();
-        String keyTarget = s3Target.getKey();
-        s3Source.getFileSystem()
-                .getClient()
-                .copyObject(CopyObjectRequest.builder()
-                        .copySource(sourceBucketName + "/" + keySource)
-                        .bucket(bucketNameTarget)
-                        .key(keyTarget)
-                        .build());
+        long objectSize = this.objectSize(s3Source);
+        long threshold = 5L * 1024L * 1024L * 1024L; //5GB
+        if (objectSize >=  threshold ){
```

Comment:
I've reformatted using the intellij standard template. Also added initial integration testing using Mockito so as to avoid the need for a real AWS environment